src/lib.rs

```diff
@@ -2081,34 +2081,42 @@ impl Build {
         Ok(())
     }
 
-    fn ios_watchos_flags(&self, cmd: &mut Tool) -> Result<(), Error> {
+    fn apple_flags(&self, cmd: &mut Tool) -> Result<(), Error> {
         enum ArchSpec {
             Device(&'static str),
             Simulator(&'static str),
             Catalyst(&'static str),
         }
 
         enum Os {
+            MacOs,
             Ios,
             WatchOs,
         }
         impl Display for Os {
             fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
                 match self {
+                    Os::MacOs => f.write_str("macOS"),
                     Os::Ios => f.write_str("iOS"),
                     Os::WatchOs => f.write_str("WatchOS"),
                 }
             }
         }
 
         let target = self.get_target()?;
-        let os = if target.contains("-watchos") {
+        let os = if target.contains("-darwin") {
+            Os::MacOs
+        } else if target.contains("-watchos") {
             Os::WatchOs
         } else {
             Os::Ios
         };
+        let is_mac = match os {
+            Os::MacOs => true,
+            _ => false,
+        };
```

Comment:
```suggestion
        let is_mac = matches!(os, Os::MacOs);
```

[`matches!`](https://doc.rust-lang.org/std/macro.matches.html) was added in 1.42, which is below the [MSRV](https://github.com/rust-lang/cc-rs/blob/57853c4bf8a89a0f4c9137eb367ac580305c6919/.github/workflows/main.yml#L113-L126).